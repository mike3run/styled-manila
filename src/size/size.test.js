import React from 'react'
import styled from 'styled-components'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { size } from '..'

test('size works with a number', () => {
  const Test = styled.div`
    ${size(20)}
  `

  const tree = renderer.create(<Test />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('width', '20px')
  expect(tree).toHaveStyleRule('height', '20px')
})

test('size works with a number', () => {
  const Test = styled.div`
    ${size(20, 45)}
  `

  const tree = renderer.create(<Test />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('width', '20px')
  expect(tree).toHaveStyleRule('height', '45px')
})

test('size works with a string', () => {
  const Test = styled.div`
    ${size('1em', '4.3em')}
  `

  const tree = renderer.create(<Test />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('width', '1em')
  expect(tree).toHaveStyleRule('height', '4.3em')
})
