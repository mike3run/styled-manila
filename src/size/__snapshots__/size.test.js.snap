// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`size works with a number 1`] = `
.c0 {
  width: 20px;
  height: 20px;
}

<div
  className="c0"
/>
`;

exports[`size works with a number 2`] = `
.c0 {
  width: 20px;
  height: 45px;
}

<div
  className="c0"
/>
`;

exports[`size works with a string 1`] = `
.c0 {
  width: 1em;
  height: 4.3em;
}

<div
  className="c0"
/>
`;
